Applications/ClientWebPortal/Source/webapp/src/utils/chainable-models/indexed-model.js

```diff
@@ -5,20 +5,14 @@ import ArrayStringKeyGenerator from 'utils/array-string-key-generator';
 export default class {
   constructor(indices, columnNames, keyGenerator = new ArrayStringKeyGenerator()) {
     // validate indices
-    let columnLength = columnNames.length;
-    if (indices == null || indices.length > columnNames) {
-      let errorMessage = 'Indices has to be an array of column numbers';
-      throw new TypeError(errorMessage);
-    } else {
-      for (let i=0; i<indices.length; i++) {
-        if (indices[i] < 0 || indices[i] >= columnNames.length) {
-          let errorMessage = 'Index ' + indices[i] + ' is out of bounds';
-          throw new RangeError(errorMessage);
-        }
+    for (let i=0; i<indices.length; i++) {
+      if (indices[i] < 0 || indices[i] >= columnNames.length) {
+        let errorMessage = 'Index ' + indices[i] + ' is out of bounds';
+        throw new RangeError(errorMessage);
       }
     }
 
-    this.indices = indices;
+    this.indices = indices.slice(0);
```

Comment:
Use:

indices.slice() instead of indices.slice(0)

Reply:
What's the reason? Does it actually do anything different? Just curious.

Reply:
Fair question.  I am very picky about attention to detail, in particular if there are two ways to accomplish the same thing, then I will always request that the simplest most minimal approach be taken.  indices.slice() does the same thing as indices.slice(0) but the former is the most minimal and simplest solution to the problem.

By explicitly passing in a parameter it suggests there is something special about that 0, that it is doing something that wouldn't otherwise be done, but in fact it's not doing anything at all so I will request that it be removed.